### PR TITLE
kubeadm: prepare for v1.9 cycle

### PR DESCRIPTION
**Pull request by andrewsykim**
**What this PR does / why we need it**:
Prepares kubeadm for v1.9 cycle which only supports v1.8 and v1.9 (when released). Deletes all v1.7.x support and sets minimum Kubernetes version to v1.8. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes 
https://github.com/kubernetes/kubeadm/issues/447
https://github.com/kubernetes/kubeadm/issues/448
https://github.com/kubernetes/kubeadm/issues/445

**Special notes for your reviewer**:
@luxas these changes seem a bit aggressive but seems reasonable given the compatibility expectations for kubeadm.

### CODE REVIEW

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -252,8 +245,5 @@ func CreateTempDirForKubeadm(dirName string) (string, error) {
 
 // GetNodeBootstrapTokenAuthGroup gets the bootstrap token auth group conditionally based on version
 func GetNodeBootstrapTokenAuthGroup(k8sVersion *version.Version) string {
```
Comment (luxas) [inline cmd/kubeadm/app/constants/constants.go]: can now remove this func and use NodeBootstrapTokenAuthGroup all the time now

**Review by luxas — COMMENTED** [review]
Basically I found one nit only, thanks! :)

**Comment by kad** [comment]
@andrewsykim are you planning to update PR to address comments ?

**Comment by andrewsykim** [comment]
@kad yes I do, was actually waiting for https://github.com/kubernetes/kubernetes/pull/53204 to go in since I would need to rebase anyways.

**Comment by luxas** [comment]
Thanks for doing that @andrewsykim, I'll get to that very soon I hope (needs fixing some tests)

**Review by luxas — APPROVED** [review]
/lgtm

Thank you, great work here!

**Comment by luxas** [comment]
@andrewsykim 
```
FAILED   hack/make-rules/../../hack/verify-bazel.sh	20s
```
Run `hack/update-bazel.sh` again please.

**Comment by andrewsykim** [comment]
Updated bazel

**Pull request merged**